docker/main/ngen/Dockerfile

```diff
@@ -783,7 +788,25 @@ FROM rocky-ngen-deps as rocky_ngen_build_testing
 #FROM rocky_build_ngen as rocky_ngen_build_testing
 #FROM ${DOCKER_INTERNAL_REGISTRY}/ngen-deps:latest as rocky_ngen_build_testing
 
+ARG NUMPY_CONSTRAINT
+
 COPY --chown=${USER} --from=rocky_init_repo ${WORKDIR}/ngen ${WORKDIR}/ngen
+
+ENV BOOST_ROOT=${WORKDIR}/boost
+
+USER root
+RUN if [ "${NGEN_WITH_PYTHON}" == "ON" ]; then \
+        chgrp -R ${USER} /usr/local/lib*/python3.* ; \
+        chmod -R g+sw /usr/local/lib*/python3.* ; \
+    fi
+USER ${USER}
+
+ENV VIRTUAL_ENV=/dmod/venv
+RUN if [ "${NGEN_WITH_PYTHON}" == "ON" ]; then python3 -m venv ${VIRTUAL_ENV} ; fi
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+RUN if [ "${NGEN_WITH_PYTHON}" == "ON" ]; then pip install numpy${NUMPY_CONSTRAINT} ; fi
```

Comment:
This is an abuse of the args parameter. Do not use this to inject python versions. Stick to setup.py and requirements.txt.